control-plane-operator/controllers/hostedcontrolplane/network_policies.go

```diff
@@ -80,46 +79,46 @@ func (r *HostedClusterReconciler) reconcileNetworkPolicies(ctx context.Context,
-		return reconcileOpenshiftMonitoringNetworkPolicy(policy, hcluster)
+		return reconcileOpenshiftMonitoringNetworkPolicy(policy, hcp)
 	}); err != nil {
 		return fmt.Errorf("failed to reconcile monitoring network policy: %w", err)
 	}
 
 	// Reconcile private-router Network Policy
-	if hcluster.Spec.Platform.Type == hyperv1.AWSPlatform {
-		policy = networkpolicy.PrivateRouterNetworkPolicy(controlPlaneNamespaceName)
-		// TODO: Network policy code should move to the control plane operator. For now,
-		// only setup ingress rules (and not egress rules) when version is < 4.14
-		ingressOnly := version.Major == 4 && version.Minor < 14
+	if hcp.Spec.Platform.Type == hyperv1.AWSPlatform {
+		policy = networkpolicy.PrivateRouterNetworkPolicy(hcp.Namespace)
 		if _, err := createOrUpdate(ctx, r.Client, policy, func() error {
-			return reconcilePrivateRouterNetworkPolicy(policy, hcluster, kubernetesEndpoint, r.ManagementClusterCapabilities.Has(capabilities.CapabilityDNS), managementClusterNetwork, ingressOnly)
+			return reconcilePrivateRouterNetworkPolicy(policy, hcp, kubernetesEndpoint, r.ManagementClusterCapabilities.Has(capabilities.CapabilityDNS), managementClusterNetwork)
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile private router network policy: %w", err)
 		}
-	} else if hcluster.Spec.Platform.Type == hyperv1.KubevirtPlatform {
-		if hcluster.Spec.Platform.Kubevirt.Credentials == nil {
+	} else if hcp.Spec.Platform.Type == hyperv1.KubevirtPlatform {
+		if hcp.Spec.Platform.Kubevirt.Credentials == nil {
 			// network policy is being set on centralized infra only, not on external infra
-			policy = networkpolicy.VirtLauncherNetworkPolicy(controlPlaneNamespaceName)
+			policy = networkpolicy.VirtLauncherNetworkPolicy(hcp.Namespace)
 			if _, err := createOrUpdate(ctx, r.Client, policy, func() error {
-				return reconcileVirtLauncherNetworkPolicy(policy, hcluster, managementClusterNetwork)
+				return reconcileVirtLauncherNetworkPolicy(policy, hcp, managementClusterNetwork)
 			}); err != nil {
 				return fmt.Errorf("failed to reconcile virt launcher policy: %w", err)
 			}
 		}
 
 		var creds *hyperv1.KubevirtPlatformCredentials
 
-		if hcluster.Spec.Platform.Kubevirt != nil && hcluster.Spec.Platform.Kubevirt.Credentials != nil {
-			creds = hcluster.Spec.Platform.Kubevirt.Credentials
+		if hcp.Spec.Platform.Kubevirt != nil && hcp.Spec.Platform.Kubevirt.Credentials != nil {
+			creds = hcp.Spec.Platform.Kubevirt.Credentials
 		}
 
-		kvInfraCluster, err := r.KubevirtInfraClients.DiscoverKubevirtClusterClient(ctx, r.Client, hcluster.Spec.InfraID, creds, hcp.Namespace, hcluster.Namespace)
+		kubevirtInfraClients := kvinfra.NewKubevirtInfraClientMap()
+
+		hclusterNamespace := manifests.HostedClusterNamespace(hcp.Namespace, hcp.Name).Name
+		kvInfraCluster, err := kubevirtInfraClients.DiscoverKubevirtClusterClient(ctx, r.Client, hcp.Spec.InfraID, creds, hcp.Namespace, hclusterNamespace)
```

Comment:
the CPO shouldn't have rbac to get from the HC namespace and therefore I'd expect this to fail otherwise that'd be a bug.